framework/src/fvkernels/FVScalarLagrangeMultiplier.C

```diff
@@ -30,13 +36,46 @@ FVScalarLagrangeMultiplier::FVScalarLagrangeMultiplier(const InputParameters & p
   : FVElementalKernel(parameters),
     _lambda_var(*getScalarVar("lambda", 0)),
     _lambda(adCoupledScalarValue("lambda")),
-    _phi0(getParam<Real>("phi0"))
+    _phi0(getParam<Real>("phi0")),
+    _constraint_type(getParam<MooseEnum>("constraint_type")),
+    _point(_constraint_type == "point-value" ? getParam<Point>("point") : Point()),
+    _my_elem(NULL)
 {
 #ifndef MOOSE_GLOBAL_AD_INDEXING
   mooseError("FVScalarLagrangeMultiplier is not supported by local AD indexing. In order to use "
              "FVScalarLagrangeMultiplier, please run the configure script in the root MOOSE "
              "directory with the configure option '--with-ad-indexing-type=global'");
 #endif
+
+  // If we have a point value constraint, we look for the mesh cell with the
+  // constraint. For this, we use a point_locator (adapted from the DiracKernels)
+  // that builds a tree to scan the mesh
+  if (_constraint_type == "point-value")
+  {
+    if (_point_locator.get() == NULL)
+    {
+      _point_locator = PointLocatorBase::build(TREE_LOCAL_ELEMENTS, _mesh);
```

Comment:
this looks like something that should be called in an intialsetup routine. like what if the mesh changes?